src/edits/performEdits.ts

```diff
@@ -50,7 +55,14 @@ export function edit_replace(text, node) {
 // except that this one takes higher-level `Edit` operations, constructed by the
 // functions: `edit_insert`, `edit_delete`, and `edit_replace`. Focus is
 // determined by the focus of the _last_ edit in `edits`.
-export function performEdits(origin, ast, edits, onSuccess=()=>{}, onError=()=>{}, annt) {
+export function performEdits(
+  origin: string,
+  ast: AST,
+  edits: Edit[],
+  onSuccess=(r:{newAST:AST, focusId:string})=>{},
+  onError=(e:any)=>{},
+  annt: string
+) {
   // Ensure that all of the edits are valid.
   //console.log('XXX performEdits:55 doing performEdits');
   for (const edit of edits) {
```

Comment:
Is this a check we elide, thanks to TypeScript? I'm guessing the answer is no, but just in case...

Reply:
I'm pretty sure the answer is no. Once the typescript is compiled to javascript, there is no type checking except explicit runtime type checking! As a library author, you may want to do runtime type checks to enforce constraints on the code that is using your library even if that code is not using typescript. It would be pretty cool though if typescript had a compilation feature where it would autogenerate runtime type checks for public interfaces to be used in a "dev mode" release.